### PR TITLE
[rush] Add ability to configure git commit messages

### LINKED ISSUE #1030: [rush] Allow customizing the commit message rush uses during publish
**Current Behavior**
Commits from Rush during `rush publish` use the string "Applying package updates." and it cannot be changed. This causes our CI system to pick up the commit and try to run another CI build.

**Desired Behavior**
The string Rush uses when committing during `rush publish` is customizable so we can add the required by our CI system to skip CI builds.

**Prior Art**
[The `command.publish.message` config option for Lerna](https://github.com/lerna/lerna#lernajson).

### PR DESCRIPTION
**Pull request by raygesualdo**
Fixes #1030.

### CODE REVIEW

File: apps/rush-lib/src/api/RushConfiguration.ts [hunk 1]
```diff
@@ -52,6 +52,7 @@ export interface IApprovedPackagesPolicyJson {
 export interface IRushGitPolicyJson {
   allowedEmailRegExps?: string[];
   sampleEmail?: string;
+  defaultCommitMessage?: string;
```
Comment (iclanton) [inline apps/rush-lib/src/api/RushConfiguration.ts]: This should be changed to `versionBumpCommitMessage`, right?

File: apps/rush-lib/src/api/RushConfiguration.ts [hunk 2]
```diff
@@ -649,6 +651,14 @@ export class RushConfiguration {
     return this._gitSampleEmail;
   }
 
+  /**
+   * [Part of the "gitPolicy" feature.]
```
Comment (iclanton) [inline apps/rush-lib/src/api/RushConfiguration.ts]: This part of the comment should be in a `@remarks` section.

File: apps/rush-lib/src/schemas/rush.schema.json [hunk 3]
```diff
@@ -119,10 +123,7 @@
             "type": "string"
           }
         },
-        "required": [
-          "variantName",
-           "description"
-        ]
+        "required": ["variantName", "description"]
```
Comment (iclanton) [inline apps/rush-lib/src/schemas/rush.schema.json]: Can you revert these formatting changes to minimize the diff?

File: apps/rush-lib/assets/rush-init/rush.json [hunk 4]
```diff
@@ -164,7 +164,13 @@
...
      * expressions.
      */
-    /*[LINE "DEMO"]*/ "sampleEmail": "mrexample@users.noreply.github.com"
+    /*[LINE "DEMO"]*/ "sampleEmail": "mrexample@users.noreply.github.com",
+
+    /**
+     * Set a default commit message to be used when Rush changes and commits files.
+     * This is helpful if you want to skip CI builds for git commits by Rush.
+     */
+    /*[LINE "DEMO"]*/ "defaultCommitMessage": "Applying package updates. [skip-ci]"
```
Comment (iclanton) [inline apps/rush-lib/assets/rush-init/rush.json]: This should be updated as well.

File: apps/rush-lib/src/api/RushConfiguration.ts [hunk 5]
```diff
@@ -649,6 +651,14 @@ export class RushConfiguration {
...
+   * A default commit message to use when committing chnages during the Rush lifecycle.
```
Comment (iclanton) [inline apps/rush-lib/src/api/RushConfiguration.ts]: I would change this to "The commit message to use when committing changes during 'rush publish'"

File: apps/rush-lib/src/api/RushConfiguration.ts [hunk 6]
```diff
@@ -203,6 +204,7 @@ export class RushConfiguration {
   // "gitPolicy" feature
   private _gitAllowedEmailRegExps: string[];
   private _gitSampleEmail: string;
+  private _gitDefaultCommitMessage: string;
```
Comment (iclanton) [inline apps/rush-lib/src/api/RushConfiguration.ts]: The type here should be `string | undefined`

**Comment by raygesualdo** [comment]
Thanks for the feedback @iclanton! I plan on updating this PR this week.

**Comment by raygesualdo** [comment]
Hey @iclanton, thanks for taking this across the finish line. Sorry I wasn't able to get back to it sooner!

**Comment by iclanton** [comment]
No worries!

**Comment by octogonz** [comment]
This was published with Rush 5.6.0.

**Pull request merged**